File: python/test/Ice/defaultServant/MyObjectI.py
# Copyright (c) ZeroC, Inc.

from generated.test.Ice.defaultServant import Test

import Ice


class MyObjectI(Test.MyObject):
    def ice_ping(self, current: Ice.Current):
        name = current.id.name

        if name == "ObjectNotExist":
            raise Ice.ObjectNotExistException()
        elif name == "FacetNotExist":
            raise Ice.FacetNotExistException()

    def getName(self, current: Ice.Current):
        name = current.id.name

        if name == "ObjectNotExist":
            raise Ice.ObjectNotExistException()
        elif name == "FacetNotExist":
            raise Ice.FacetNotExistException()

        return name

File: js/test/Ice/defaultServant/MyObjectI.ts
// Copyright (c) ZeroC, Inc.

import { Ice } from "@zeroc/ice";
import { Test } from "./Test.js";

export class MyObjectI extends Test.MyObject {
    ice_ping(current: Ice.Current) {
        const name = current.id.name;

        if (name == "ObjectNotExist") {
            throw new Ice.ObjectNotExistException();
        } else if (name == "FacetNotExist") {
            throw new Ice.FacetNotExistException();
        }
    }

    getName(current: Ice.Current): string {
        const name = current.id.name;

        if (name == "ObjectNotExist") {
            throw new Ice.ObjectNotExistException();
        } else if (name == "FacetNotExist") {
            throw new Ice.FacetNotExistException();
        }
        return name;
    }
}
